src/components/frame/Extensions/pages/ExtensionDetailsPage/hooks.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useCallback, useMemo } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
+
+import { ColonyVersion, Extension, ExtensionVersion, isExtensionCompatible } from '@colony/colony-js';
+import { useAsyncFunction, useColonyContext } from '~hooks';
+import { ActionTypes } from '~redux';
+import { AnyExtensionData } from '~types';
+import { mapPayload } from '~utils/actions';
+import { MIN_SUPPORTED_COLONY_VERSION } from '~constants';
+import Toast from '~shared/Extensions/Toast';
+
+export const useGetExtensionsViews = async () => {
+  const colonyMetrics = 'https://api.thegraph.com/subgraphs/name/arrenv/colony-metrics-subgraph';
+  const metricsRes = fetch(colonyMetrics, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      query: `{
+          votingReputationExtensions(first: 5) {
+            installs
+          }
+          oneTxPaymentExtensions(first: 5) {
+            installs
+          }
+        }`,
+      variables: {},
+    }),
+  })
+    .then((response) => response.json())
+    .then((data) => {
+      return data;
+    })
+    .catch((e) => {
+      console.error(e);
+    });
+
+  const [metricsOutput] = await Promise.all([metricsRes]);
+  return metricsOutput.data;
+};
+
+export const useExtensionDetailsPage = (extensionData: AnyExtensionData) => {
+  const { colony } = useColonyContext();
+  const navigate = useNavigate();
+
+  const extensionValues = useMemo(() => {
+    return {
+      colonyAddress: colony?.colonyAddress,
+      extensionData,
+    };
+  }, [colony?.colonyAddress, extensionData]);
+
+  const submitInstall = ActionTypes.EXTENSION_INSTALL;
+  const errorInstall = ActionTypes.EXTENSION_INSTALL_ERROR;
+  const successInstall = ActionTypes.EXTENSION_INSTALL_SUCCESS;
+
+  const submitUpgrade = ActionTypes.EXTENSION_UPGRADE;
+  const errorUpgrade = ActionTypes.EXTENSION_UPGRADE_ERROR;
+  const successUpgrade = ActionTypes.EXTENSION_UPGRADE_SUCCESS;
+
+  const transform = mapPayload(() => ({
+    colonyAddress: colony?.colonyAddress,
+    extensionId: extensionData.extensionId,
+    version: extensionData.availableVersion,
+  }));
+
+  const asyncFunctionInstall = useAsyncFunction({
+    submit: submitInstall,
```

Comment:
submit: ActionTypes.EXTENSION_INSTALL,
error: ActionTypes.EXTENSION_INSTALL_ERROR,
success: ActionTypes.EXTENSION_INSTALL_SUCCESS

please do the same with upgrate action types

Reply:
Done